megamek/src/megamek/server/AutosaveService.java

```diff
@@ -35,20 +34,21 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
-public class AutosaveService {
-    public final String FILENAME_FORMAT = "Round-%d-autosave%s.sav.gz";
+class AutosaveService {
 
-    //region Constructors
-    public AutosaveService() {
+    public static final String FILENAME_FORMAT = "Round-%d-autosave%s.sav.gz";
 
+    private final AbstractGameManager gameManager;
+
+    AutosaveService(AbstractGameManager gameManager) {
+        this.gameManager = gameManager;
```

Comment:
There was a reason I didn't do this for the rolling autosaves, but for the life of me I can't recall why.  Maybe because the gameManager wasn't guaranteed to stay the same for the life of the service?

Reply:
Having difficulty imagining any reason. Guess we'll have to see.